tools/ci_build/github/azure-pipelines/templates/jobs/download_linux_qnn_sdk.yml

```diff
@@ -21,16 +21,18 @@ steps:
       sdk_file="$(QnnSDKRootDir)/sdk.yaml"
       qnn_sdk_version=$(grep '^version:' "$sdk_file" | head -n 1 | cut -d':' -f2 | xargs | cut -d'.' -f1-3)
       if [[ -n "$qnn_sdk_version" ]]; then
-        echo "##vso[task.setvariable variable=QnnSDKVersionYaml;isOutput=true]$qnn_sdk_version"
+        # Do not output ##vso[] commands with `set -x` or they may be parsed again and include a trailing quote.
+        set +x
+        echo "##vso[task.setvariable variable=QnnSDKExtractedVersion ;isOutput=true]$qnn_sdk_version"
       else
         echo "Version not found in sdk.yaml."
         exit 1
       fi
-    displayName: Set_QnnSDKVersionYaml based on sdk.yaml
-    name: Set_QnnSDKVersionYaml
+    displayName: Get_QnnSDKExtractedVersion based on sdk.yaml
+    name: Get_QnnSDKExtractedVersion
```

Comment:
Out of interest, if we don't set 'name:' here does the prefix to the variable go away and usage of this value would be `QnnSDKExtractedVersion ` instead of `Get_QnnSDKExtractedVersion.QnnSDKExtractedVersion` ? 

I see other usages of task.setvariable in CIs where that seems to be the case.

Reply:
The prefix of task name is not required if the variable is used in the same job. To reference a variable from a task from a different job, it's `[dependencies.JOB.outputs['TASK.VARIABLE']]`(https://learn.microsoft.com/en-us/azure/devops/pipelines/process/variables?view=azure-devops&tabs=yaml%2Cbatch#use-output-variables-from-tasks). If it's across stages, then `stageDependencies.STAGE.JOB.outputs['TASK.VARIABLE']`